coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -483,10 +489,53 @@ class MultiPartitionPlanner(val partitionLocationProvider: PartitionLocationProv
     }
   }
 
+  /**
+   * The only filters expected in this planner are Equals and pipe-concatenated EqualsRegex.
+   * This method resolves any filter groups with pipe-concatenated filters into sets of Equals-only filters.
+   *
+   * Example: [{A=~1|2, B=1}, {A=1, B=2}] -> [{A=1, B=1}, {A=2, B=1}, {A=1, B=2}]
+   */
+  private def getNonMetricShardKeyFilters(lp: LogicalPlan): Seq[Seq[ColumnFilter]] = {
```

Comment:
should we have a targeted unit test just for this function ?

Reply:
Test added ✅